presentation/src/main/java/team/jsv/icec/base/BaseActivity.kt

```diff
@@ -0,0 +1,30 @@
+package team.jsv.icec.base
+
+import android.os.Bundle
+import android.widget.Toast
+import androidx.annotation.LayoutRes
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+
+abstract class BaseActivity<T : ViewDataBinding>(
+    @LayoutRes private val layoutId: Int
+) : AppCompatActivity() {
+
+    protected lateinit var binding: T
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView(this, layoutId)
+        binding.lifecycleOwner = this
```

Comment:
```
Sets the LifecycleOwner that should be used for observing changes of LiveData in this binding. If a LiveData is in one of the binding expressions and no LifecycleOwner is set, the LiveData will not be observed and updates to it will not be propagated to the UI. When using Data Binding with Fragments, make sure to use Fragment.getViewLifecycleOwner(). Using the Fragment as the LifecycleOwner might cause memory leaks since the Fragment's Lifecycle outlives the view Lifecycle. When using Data Binding with Activities, you can use the Activity as the LifecycleOwner.
Params:
lifecycleOwner – The LifecycleOwner that should be used for observing changes of LiveData in this binding
```